pandas/core/reshape.py

```diff
@@ -1221,17 +1293,23 @@ def get_empty_Frame(data, sparse):
         else:
             return SparseDataFrame(index=index, default_fill_value=0)
 
-    # if all NaN
-    if not dummy_na and len(levels) == 0:
+    # If we get all NaN and are not making a dummy col, then just return.
+    # GH15826
+    if len(levels) == 0 and not dummy_na:
         return get_empty_Frame(data, sparse)
 
+    # Record NaN values before we munge the codes, GH15826
+    nan_codes_msk = codes == -1
```

Comment:
For future notes, you have to cache the nan-values since anything with `code == -1` will get overwritten if you pass `dummy_na=True`.